device/core/test/_client_test.js

```diff
@@ -864,6 +864,59 @@ describe('Client', function () {
      client.getTwin(done, fakeTwin);
     });
   });
+
+  describe('setRetryPolicy', function() {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_083: [The `setRetryPolicy` method shall throw a `ReferenceError` if the policy object is falsy.]*/
+    [null, undefined, ''].forEach(function(badPolicy) {
+      it('throws a ReferenceError if policy is \'' + badPolicy + '\'', function() {
+        var client = new Client(new EventEmitter());
+        assert.throws(function() {
+          client.setRetryPolicy(badPolicy);
+        }, ReferenceError);
+      });
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_084: [The `setRetryPolicy` method shall throw an `ArgumentError` if the policy object doesn't have a `shouldRetry` method.]*/
+    it('throws an ArgumentError if the policy does not have a shouldRetry method', function() {
+      var client = new Client(new EventEmitter());
+      var badPolicy = {
+        nextRetryTimeout: function() {}
+      };
+      assert.throws(function() {
+        client.setRetryPolicy(badPolicy);
+      }, errors.ArgumentError);
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_085: [The `setRetryPolicy` method shall throw an `ArgumentError` if the policy object doesn't have a `nextRetryTimeout` method.]*/
+    it('throws an ArgumentError if the policy does not have a nextRetryTimeout method', function() {
+      var client = new Client(new EventEmitter());
+      var badPolicy = {
+        shouldRetry: function () {}
+      };
+      assert.throws(function() {
+        client.setRetryPolicy(badPolicy);
+      }, errors.ArgumentError);
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_086: [Any operation happening after a `setRetryPolicy` call should use the policy set during that call.]*/
```

Comment:
>Any operation [](start = 44, length = 13)

debt for later: the SRS says "any operation" but you're only testing sendEvent.

Reply:
yeah and I was thinking sendEvent just as an example but it's true we may want to test all operations.. hmmm.

---
In reply to: [145965518](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r145965518) [](ancestors = 145965518)

Reply:
I think this can wait. (famous last words, I know...)

---
In reply to: [146044146](https://github.com/Azure/azure-iot-sdk-node/pull/114#discussion_r146044146) [](ancestors = 146044146,145965518)